test/manipulation.test.js

```diff
@@ -193,6 +297,29 @@ describe('manipulation', function () {
       });
     });
 
+    it('should create batch of objects with beforeCreate (promise variant)', function(done) {
+      Person.beforeCreate = function(next, data) {
+        if (data && data.name === 'A') {
+          return next(null, {id: 'a', name: 'A'});
+        } else {
+          return next();
+        }
+      };
+      var batch = [
+        {name: 'A'},
+        {name: 'B'},
+        undefined
+      ];
+      Person.create(batch)
+      .then(function(ps) {
+        should.exist(ps);
+        ps.should.be.instanceOf(Array);
+        ps.should.have.lengthOf(batch.length);
+        ps[0].should.be.eql({id: 'a', name: 'A'});
+        done();
+      }).catch(done);
+    });
```

Comment:
> I think I just duplicated every test that I could rewrite using promises
rather than callbacks... we can probably take out a bunch of them.

So I've noticed a lot of talk about whitespace which I'd like to point out as basically annoying but could really care less if fixed before this PR lands... but this (duplicating a large number of tests) is concerning. @bajtos @raymondfeng any ideas on how to test promise support without duplicating so many tests?